cruise-control-core/src/main/java/com/linkedin/cruisecontrol/monitor/sampling/aggregator/MetricSampleAggregator.java

```diff
@@ -217,9 +217,11 @@ public MetricSampleAggregationResult<G, E> aggregate(long from, long to, Aggrega
       MetricSampleAggregationResult<G, E> result = new MetricSampleAggregationResult<>(generation(), completeness);
       Set<E> entitiesToInclude =
           interpretedOptions.includeInvalidEntities() ? interpretedOptions.interestedEntities() : completeness.validEntities();
+      LOG.debug("Including {} entities during metric aggregation.", entitiesToInclude.size());
       for (E entity : entitiesToInclude) {
         RawMetricValues rawValues = _rawMetrics.get(entity);
         if (rawValues == null) {
+          LOG.debug("Failed to find entity {} from _rawMetrics.", entity);
```

Comment:
IMO, would be better to log it as a WARN

Reply:
Hi @bsandeep23 , I originally added a WARN level log here, but then after thinking it deeper I think it may be safer to make it debug for now.

The reason is we currently don't clearly know whether this is a legit path or not, and how frequently this happens. I would worry about this log line will be hit as hot path for some cases which could cause logging pollution. So for now I think it's probably safer to make it debug level.